### PR TITLE
Use latest version of ansible-operator

### PR DESCRIPTION
**Pull request by ahadas**
Update ansible-operator to latest version, this requires
- Updating kustomize
- Since requirements are no longer installed implicitly, the build process for the image operator was switched to a Containerfile to install the ansible collections manually

NOTE:
* This does not have the issue that was seen downstream
* Building the operator only works with docker now

### CODE REVIEW

File: operator/Containerfile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+FROM quay.io/operator-framework/ansible-operator:main
```
Comment (bennyz) [inline operator/Containerfile:1]: need to pin this to the release instead of main

File: WORKSPACE [hunk 2]
```diff
@@ -3341,8 +3349,7 @@ container_pull(
 
 container_pull(
     name = "ansible-operator-image",
-    # v1.22.0
-    digest = "sha256:e07ba82127e76f282cb61fad6cfd990ab137533e5e996686576dec088d5e7e44",
+    tag = "v1.34.1",
```
Comment (ahadas) [inline WORKSPACE:3352]: need to remove

File: operator/Containerfile (hunk 1 above)
Comment (ahadas) [reply operator/Containerfile:1]: I'm good with this!

**Comment by ahadas** [comment]
related to #713

**Comment by bennyz** [comment]
I can't build this locally with podman:
```
Error: payload does not match any of the supported image formats:
 * oci: open /var/tmp/podman2879577425/index.json: not a directory
 * oci-archive: loading index: open /var/tmp/container_images_oci2212243706/index.json: no such file or directory
 * docker-archive: Some layer tarfiles are missing in the tarball
 * dir: open /var/tmp/podman2879577425/manifest.json: not a directory
make: *** [Makefile:189: build-operator-image] Error 125
```

**Review by ahadas — COMMENTED** [review]
looks good but please explain the changes in the description of the PR

**Review by bennyz — APPROVED** [review]

**Pull request merged**